src/servers/src/http/influxdb.rs

```diff
@@ -56,7 +77,9 @@ pub async fn influxdb_write_v1(
         .map(|val| parse_time_precision(val))
         .transpose()?;
 
-    influxdb_write(&db, precision, lines, handler, query_ctx).await
+    let gzip = is_gzip(&headers)?;
```

Comment:
After some consideration, I believe it would be beneficial to implement the feature as an Axum middleware. Please refer to https://docs.rs/axum/latest/axum/middleware/index.html. This is because other HTTP protocols, such as `/sql` and Opentsdb, may also encounter similar issues.

Reply:
Actually there is a `RequestDecompressionLayer` in tower-http. Let me try to leverage it.

Reply:
See https://github.com/GreptimeTeam/greptimedb/pull/3494#issuecomment-1994350071